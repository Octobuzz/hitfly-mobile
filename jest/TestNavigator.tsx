// https://github.com/react-navigation/rfcs/issues/75
import React, { Children } from 'react'
import {
  createAppContainer,
  createSwitchNavigator,
  NavigationParams,
} from 'react-navigation'

const TestNavigator = ({
  children,
  params,
}: {
  children: any
  params?: NavigationParams
}) => {
  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: { screen: () => Children.only(children), params },
    }),
  )
  return <Navigator />
}

export default TestNavigator
