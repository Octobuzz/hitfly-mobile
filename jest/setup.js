import 'jest-styled-components'
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock'

jest.mock('react-native-vector-icons/Foundation', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')
jest.mock('react-native-vector-icons/Feather', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')
jest.mock('react-native-vector-icons/SimpleLineIcons', () => 'Icon')

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)

jest.mock('react-native-track-player', () => {
  return {
    addEventListener: jest.fn(),
    registerEventHandler: jest.fn(),
    registerPlaybackService: jest.fn(),
    setupPlayer: jest.fn(),
    destroy: jest.fn(),
    updateOptions: jest.fn(),
    add: jest.fn(),
    remove: jest.fn(),
    skip: jest.fn(),
    skipToNext: jest.fn(),
    skipToPrevious: jest.fn(),
    removeUpcomingTracks: jest.fn(),
    // playback commands
    reset: jest.fn(),
    play: jest.fn(),
    pause: jest.fn(),
    stop: jest.fn(),
    seekTo: jest.fn(),
    setVolume: jest.fn(),
    setRate: jest.fn(),
    // player getters
    getQueue: jest.fn(),
    getTrack: jest.fn(),
    getCurrentTrack: jest.fn(),
    getVolume: jest.fn(),
    getDuration: jest.fn(),
    getPosition: jest.fn(),
    getBufferedPosition: jest.fn(),
    getState: jest.fn(),
    getRate: jest.fn(),
  }
})

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})
