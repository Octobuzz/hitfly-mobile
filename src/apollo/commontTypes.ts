export type CollectionsType = 'recommended' | 'musicFan'
