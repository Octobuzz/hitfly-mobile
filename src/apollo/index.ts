import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
