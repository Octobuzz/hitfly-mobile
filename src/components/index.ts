export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
