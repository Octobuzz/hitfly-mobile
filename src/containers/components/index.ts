export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
