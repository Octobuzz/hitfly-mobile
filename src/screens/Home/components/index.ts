export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
