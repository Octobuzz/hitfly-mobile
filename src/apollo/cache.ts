import { InMemoryCache } from 'apollo-cache-inmemory'
import { persistCache } from 'apollo-cache-persist'
import { storageInstance } from 'src/utils'
import gql from 'graphql-tag'

export const defaults = {
  headerSettings: {
    mode: 'dark',
    state: 'main',
    __typename: 'HeaderSettings',
  },
  isPlaying: false,
  activeTrack: null,
  playlist: [],
}

export const typeDefs = gql`
  type Track {
    id: Int!
    title: String
    singer: String
    date: String
    songText: String
    fileUrl: String
    musicWave: [Int]
    length: Int
  }
  type HeaderSettings {
    style: String!
    state: String!
  }
`

export default async (): Promise<InMemoryCache> => {
  const cache = new InMemoryCache({
    freezeResults: true,
  })

  await persistCache({
    cache,
    debug: __DEV__,
    // @ts-ignore
    storage: storageInstance,
  })

  return cache
}
