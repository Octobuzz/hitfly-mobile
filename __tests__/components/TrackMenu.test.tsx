import { render } from '../../jest/test-utils'
import { TrackMenu } from 'src/components'
import React from 'react'
import { Track } from '../../src/apollo'
import { fireEvent } from '@testing-library/react-native'

describe('TrackMenu', () => {
  const track = {
    title: 'title',
    id: 2,
    cover: [{}],
  } as Track

  const props = {
    onPressLike: jest.fn(),
    onPressEdit: jest.fn(),
    onPressCancel: jest.fn(),
    track,
  }

  it('renders correctly', () => {
    const { asJSON } = render(<TrackMenu />)
    expect(asJSON()).toMatchSnapshot()
  })
  it('renders correctly with props', () => {
    const { asJSON, getAllByRole, getByText } = render(<TrackMenu {...props} />)

    fireEvent.press(getAllByRole('summary')[0])
    fireEvent.press(getAllByRole('summary')[1])
    fireEvent.press(getByText('Отмена'))
    expect(asJSON()).toMatchSnapshot()
  })
})
