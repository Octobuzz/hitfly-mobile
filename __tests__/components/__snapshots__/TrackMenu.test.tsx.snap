// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackMenu renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`TrackMenu renders correctly 2`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`TrackMenu renders correctly with handlers 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    noFill={true}
    paddingBottom={32}
    style={
      Array [
        Object {
          "paddingBottom": 32,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        },
      ]
    }
    testID="like"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "borderRadius": 4,
                "height": 32,
                "width": 32,
              },
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="cover"
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginHorizontal": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "GothamPro-Bold",
                "fontSize": 12,
                "lineHeight": 14,
              },
            ]
          }
        >
          title
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "GothamPro",
                "fontSize": 10,
                "lineHeight": 12,
                "opacity": 0.6,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.5)",
            "height": 1,
            "marginBottom": 34,
            "marginTop": 24,
          },
        ]
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 34,
          },
        ]
      }
      testID="like"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            Array [
              Object {
                "marginRight": 11,
                "textAlign": "center",
                "width": 32,
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "GothamPro",
              "fontSize": 14,
              "lineHeight": 14,
            },
          ]
        }
      >
        Понравилось
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 34,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            Array [
              Object {
                "marginRight": 11,
                "textAlign": "center",
                "width": 32,
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "GothamPro",
              "fontSize": 14,
              "lineHeight": 14,
            },
          ]
        }
        testID="like"
      >
        Редактировать
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "marginHorizontal": 20,
          },
        ]
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4282554366,
            4291060189,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 28,
              "paddingBottom": 1,
              "paddingLeft": 1,
              "paddingRight": 1,
              "paddingTop": 1,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "#313131",
                "borderRadius": 28,
                "color": "#ffffff",
                "fontFamily": "GothamPro",
                "fontSize": 16,
                "lineHeight": 20,
                "overflow": "hidden",
                "paddingBottom": 20,
                "paddingLeft": 29,
                "paddingRight": 29,
                "paddingTop": 20,
                "textAlign": "center",
              },
            ]
          }
          type="outline-black"
        >
          Отмена
        </Text>
      </BVLinearGradient>
    </TouchableOpacity>
  </View>
</View>
`;
